velox/experimental/wave/dwio/decode/GpuDecoder-inl.cuh

```diff
@@ -256,29 +260,57 @@ __device__ inline void decodeDictionaryOnBitpack(GpuDecode& plan) {
   }
 }
 
+template <int32_t kBlockSize>
+struct BlockScanStorage {
```

Comment:
Can this be inferred from `BlockScanT`?

Reply:
We can do something more precise for each stage where a scan is needed. I tried to keep it as consistent with existing code as possible in this PR, where we use a maximum size required even if the stage might technically use less. Let me know if this is something that we should improve as part of this change or OK to do as a follow up?